p2p/switch.go

```diff
@@ -485,9 +484,10 @@ func (s *Switch) processMessage(ime net.IncomingMessageEvent) {
 
 // RegisterDirectProtocolWithChannel registers a direct protocol with a given channel. NOTE: eventually should replace RegisterDirectProtocol
 func (s *Switch) RegisterDirectProtocolWithChannel(protocol string, ingressChannel chan service.DirectMessage) chan service.DirectMessage {
-	s.protocolHandlerMutex.Lock()
+	if s.started == 1 {
+		panic("Attempting to register after p2p has started")
```

Comment:
same here

Reply:
also I think its worth to either unify the message or include some info about what is the origin of the panic